assembl/tests/views_tests/test_graphql.py

```diff
@@ -1631,3 +1631,233 @@ def test_query_discussion_sentiments_count(
+            image {
+                externalUrl
+                title
+            }
+            doc {
+                externalUrl
+                title
+            }
+        }
+    }
+}
+""" % (title_entries, text_entries, embed_code), context_value=graphql_request, variable_values={"img": u"variables.img", "doc": u"variables.doc"})
+    result = res.data
+    assert result is not None
+    assert result['createResource'] is not None
+    resource = result['createResource']['resource']
+    assert resource['title'] == u'Première ressource'
+    assert resource['text'] == u'Lorem ipsum dolor sit amet, consectetur adipisicing elit.'
+    assert resource['embedCode'] == u'iframe foobar'
+
+    assert '/documents/' in resource['image']['externalUrl']
+    assert resource['image']['title'] == 'img.png'
+
+    assert '/documents/' in resource['doc']['externalUrl']
+    assert resource['doc']['title'] == 'mydoc.pdf'
+
+
+def test_delete_resource(graphql_request, resource):
+    resource_id = to_global_id('Resource', resource.id)
+    res = schema.execute(u"""
+mutation deleteResource {
+    deleteResource(
+        resourceId:"%s",
+    ) {
+        success
+    }
+}
+""" % resource_id, context_value=graphql_request)
+    assert res.data['deleteResource']['success'] is True
+    res = schema.execute(u'query { resources { id } }', context_value=graphql_request)
+    assert json.loads(json.dumps(res.data)) == {u'resources': []}
+
+
+def test_update_resource(graphql_request, resource_with_image_and_doc):
+    resource = resource_with_image_and_doc
+    resource_id = to_global_id('Resource', resource.id)
+
+    import os
+    from io import BytesIO
+
+    class FieldStorage(object):
+        file = BytesIO(os.urandom(16))
+
+        def __init__(self, filename, type):
+            self.filename = filename
+            self.type = type
+
+    graphql_request.POST['variables.img'] = FieldStorage(
+        u'path/to/new-img.png', 'image/png')
+    graphql_request.POST['variables.doc'] = FieldStorage(
+        u'path/to/new-doc.pdf', 'application/pdf')
+
+    res = schema.execute(u"""
+mutation updateResource($img:String,$doc:String) {
+    updateResource(
+        id: "%s",
+        titleEntries:[
+            {value:"My resource", localeCode:"en"},
+            {value:"Ma ressource", localeCode:"fr"}
+        ],
+        textEntries:[
+            {value:"Text in english", localeCode:"en"},
+            {value:"Texte en français", localeCode:"fr"}
+        ],
+        embedCode:"nothing",
+        image:$img,
+        doc:$doc
+    ) {
+        resource {
+            title(lang:"fr")
+            text(lang:"fr")
+            embedCode
+            image {
+                externalUrl
+                title
+            }
+            doc {
+                externalUrl
+                title
+            }
+        }
+    }
+}
+""" % (resource_id), context_value=graphql_request, variable_values={"img": u"variables.img", "doc": u"variables.doc"})
```

Comment:
@cedricmessiant for new graphql tests you shouldn't use python variables interpolations and only use variable_values.

Reply:
done